### PR TITLE
Adding unit test for PnP solver 

### PR DESCRIPTION
**Pull request by MikiyaShibuya**
The current PnP solver takes care of coordinates or some vector as arrays.
So, I will replace them to Eigen::Vectors.

Firstly, I added a unit test to check if the solver estimates camera-pose correctly.

### CODE REVIEW

File: test/openvslam/solve/pnp_solver.cc [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+double clamp(const double x, const double min_val, const double max_val) {
+    assert(min_val < max_val);
+    if (x > max_val)
+        return max_val;
+    else if (x < min_val)
+        return min_val;
+    else
+        return x;
+}
+double angle_of_rot(const Mat33_t& rot) {
```
Comment (shinsumicco) [inline test/openvslam/solve/pnp_solver.cc]: Can be replaced with norm of `converter::to_angle_axis(rot)` ?

File: test/openvslam/solve/pnp_solver.cc [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    // Create bearing vectors from pose and landmarks
+    eigen_alloc_vector<Vec3_t> bearings;
+    create_bearing_vectors(rot_gt, trans_gt, landmarks, bearings);
+
+    // Make dummy keypoints
+    std::vector<cv::KeyPoint> keypts;
+    std::vector<float> scale_factors;
+    for (unsigned int i = 0; i < num_landmarks; i++) {
+        cv::KeyPoint kp;
+        keypts.push_back(kp);
```
Comment (shinsumicco) [inline test/openvslam/solve/pnp_solver.cc]: Can be replaced as follows?

```cpp
        keypts.emplace_back(cv::KeyPoint{});
```

And could you insert the comment which states the default value of `octave` contained in `cv::KeyPoint` [is 0](https://docs.opencv.org/3.4/d2/d29/classcv_1_1KeyPoint.html#a8ce2d246f9dbd4ce3f1539bf40be9533) ?

File: test/openvslam/solve/pnp_solver.cc (hunk 1 above)
Comment (MikiyaShibuya) [reply test/openvslam/solve/pnp_solver.cc]: It is possible. I missed that function.

File: test/openvslam/solve/pnp_solver.cc (hunk 2 above)
Comment (MikiyaShibuya) [reply test/openvslam/solve/pnp_solver.cc]: Yes, your suggestion is more efficient.

File: test/openvslam/solve/pnp_solver.cc [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+#include "openvslam/solve/pnp_solver.h"
+#include "openvslam/type.h"
+#include <gtest/gtest.h>
+#include <opencv2/core/types.hpp>
+#include <memory>
+#include <algorithm>
+
+using namespace openvslam;
+
+double clamp(const double x, const double min_val, const double max_val) {
```
Comment (shinsumicco) [inline test/openvslam/solve/pnp_solver.cc]: This function is not used.

Comment (MikiyaShibuya) [reply test/openvslam/solve/pnp_solver.cc]: The function is no longer used. I removed it.

**Comment by shinsumicco** [comment]
LGTM. Thanks!

**Pull request merged**